std/functional.d

```diff
@@ -1814,14 +1813,33 @@ if (isCallable!(F))
     {
         return fp;
     }
+    else static if (is(F Func == Func*) && is(Func == function) && is(Func Params == __parameters))
+    {
+        // https://issues.dlang.org/show_bug.cgi?id=24007 - cannot specify linkage on function literal:
+        //alias dg = delegate(Params params) const => F.init(params);
+        //typeof(dg) result;
+		// Workaround:
+        struct S
+        {
+            mixin("extern(", __traits(getLinkage, fp), ") auto ref dg(Params params) const => F.init(params);");
+        }
+        typeof(&S().dg) result; // inlining `dg` infers attributes incorrectly
+
+        () @trusted
+        {
+            // assigning funcptr is @system, but it’s safe here because `fp` needs no context
```

Comment:
So, what did we expect the compiler to do that makes this delegate call a valid function call?